### PR TITLE
Issue 1730 mesh with missing vertices

### PR DESCRIPTION
**Pull request by doutriaux1**
@aashish24 @sankhesh thanks to @sankhesh help I got the shapes to be drawn correctly, but they still appear to be drawn as missing, I think the "nan" bit throws off VTK ? Help wanted :wink:

### CODE REVIEW

**Comment by aashish24** [comment]
great @doutriaux1 how should we handle nan?

**Comment by doutriaux1** [comment]
@aashish24 basically nan means `ignore this vertice` that's how we handle grids which have some cells with a different number of vertices between them (3 vs 4 here) we use the greatest number of vertices and set the unused ones to missing, which @sankhesh suggested we convert to `nan` for VTK. Unfortunately VTK does pick the correct shape now but seems to mark the cell has `bad`

**Comment by doutriaux1** [comment]
@aashish24 @sankhesh @chaosphere2112 ready for review

**Comment by doutriaux1** [comment]
goes with: https://github.com/UV-CDAT/uvcdat-testdata/pull/94

**Comment by aashish24** [comment]
@doutriaux1 if this is a uniform grid, we should probably using masking array. Thoughts?

**Comment by chaosphere2112** [comment]
@aashish24 Actually, this is for `meshfill`, which is inherently a non-uniform grid.

**Comment by doutriaux1** [comment]
@aashish24 it is actually using masked arrays anyway. And @chaosphere2112 is right these are non-uniform grids, and in this case really non-uniform (different number of vertices for each cell)

**Comment by aashish24** [comment]
@doutriaux1 thanks.. can you fix the style issue?

**Comment by doutriaux1** [comment]
nah...

**Comment by aashish24** [comment]
@chaosphere2112 sorryI meant to say regular grid / structured grid which curve linear is.

File: Packages/vcs/Lib/vcs2vtk.py [hunk 1]
```diff
@@ -263,7 +263,7 @@ def genGrid(data1, data2, gm, deep=True, grid=None, geo=None):
                 xM = m[:, 1].max()
                 ym = m[:, 0].min()
                 yM = m[:, 0].max()
-                N = m.shape[0]
+                numberOfCells = m.shape[0]
```
Comment (aashish24) [inline Packages/vcs/Lib/vcs2vtk.py]: nice var name thanks :+1:

**Comment by aashish24** [comment]
@doutriaux1 changes LGTM :+1:

**Comment by aashish24** [comment]
If bots are happy, I am going to merge it.. nice job @doutriaux1 @sankhesh

**Comment by doutriaux1** [comment]
@aashish24 please do not forget to merge baselines as well. Thanks. I just merged it in. Thx @durack1 for spotting this.

**Pull request merged**